lib/java/desktop-environment/src/main/java/org/enso/desktopenvironment/MacTrashBin.java

```diff
@@ -0,0 +1,85 @@
+package org.enso.desktopenvironment;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import org.graalvm.nativeimage.UnmanagedMemory;
+import org.graalvm.nativeimage.c.CContext;
+import org.graalvm.nativeimage.c.function.CFunction;
+import org.graalvm.nativeimage.c.type.CCharPointer;
+import org.graalvm.nativeimage.c.type.CTypeConversion;
+import org.graalvm.word.Pointer;
+import org.graalvm.word.WordFactory;
+import org.slf4j.LoggerFactory;
+
+@CContext(MacTrashBin.CoreServices.class)
+final class MacTrashBin implements TrashBin {
+
+  @CFunction
+  static native int FSPathMakeRefWithOptions(
+      CCharPointer path, int flags, Pointer buffer, Pointer any);
+
+  @CFunction
+  static native int FSMoveObjectToTrashSync(Pointer source, Pointer target, int flags);
+
+  @Override
+  public boolean isSupported() {
+    return true;
+  }
+
+  @Override
+  public boolean moveToTrash(Path path) {
+    if (Platform.getOperatingSystem().isMacOs())
```

Comment:
I'd say this check is unnecessary because the class is not exposed, and its instantiation depends on the `Platform.getOperatingSystem()`. I.e. the factory method guarantees the correct OS

Reply:
It would fail native image build otherwise.